[Feat(#90)] AWS S3 음성파일 적재 기능

jaepyo-Lee opened the pull request:
## #️⃣연관된 이슈
#90 

## 📝작업 내용
- S3 음성파일 적재기능 추가


## 💬리뷰 요구사항(선택)
- git은 참 어렵네요.. rebase 사용했다가 뭔가 꼬인거같아서 혹시 이상한것 있으면 말씀해주세요!

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* **New Features**
  * Added support for uploading audio files to AWS S3 storage when submitting user speech.
  * Introduced integration with AWS S3, allowing secure and scalable file storage.
* **Chores**
  * Updated dependencies to include AWS S3 integration libraries.
  * Added configuration for AWS S3 client setup.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

Commit:
<feat #83> 오늘의 미나리 1일1회 제한 로직 추가

Commit:
<feat #83> 1일 1회 제한 로직 추가

- Answer의 success 필드 삭제
- 상태 필드 값 수정
- 프론트 개발상 불편으로 제한 로직 임시주석
- 스웨거에 응답값 보이지 않는 오류 수정

Commit:
<feat #83> stt변환시 음성파일 시간과 면접일자 추가

- Answer에 면접일자와 면접진행시간 필드 추가

Commit:
<feat #83> stt변환 응답값 스웨거 설명 추가

Commit:
<feat #83> 응답값 스웨거 설명 수정

Commit:
<fix #83> log 단계 수정

Commit:
feat: aws s3업로드 추가

Commit:
feat: aws s3업로드 추가

# Conflicts:
#	build.gradle

Commit:
remove: yml파일 추적무시

Commit:
Merge branch 'feat/#80' of https://github.com/prography/10th-3team-minari-BE into feat/#90

# Conflicts:
#	build.gradle

Commit:
<fix #90> s3 업로드 경로수정

Commit:
<fix #90> s3 업로드 경로수정

Commit:
Merge branch 'feat/#90' of https://github.com/prography/10th-3team-minari-BE into feat/#90

isayaksh commented on src/main/java/com/prography/minari/batch/item/processor/MailContentCreateProcessor.java:
```diff
@@ -0,0 +1,41 @@
+    public ItemProcessor<User, MailRequest> processor() {
+        return user -> {
+            Optional<Question> questionOpt = questionReader.readDaily(user, List.of(Domain.CS, user.getDomain()), user.getDaysSinceJoined());
+            if(questionOpt.isPresent()) {
+                Question question = questionOpt.get();
+                return mailTemplateCreater.create("오늘의 미나리",
+                        user.getEmail(),
+                        Map.of("category",question.getDomain().toString(),"keyword",question.getTags().getFirst()),
+                        "today-minari");
+            }
```
isPresent 대신에 orElse를 한번 사용해보시겠어요?

isayaksh commented:
무엇인가가 꼬인 것 같은데..

Commit:
Merge branch 'develope' of https://github.com/prography/10th-3team-minari-BE into feat/#90

# Conflicts:
#	build.gradle
#	src/main/java/com/prography/minari/mail/service/MailService.java

Commit:
conflict merge

coderabbitai[bot] committed:
📝 Add docstrings to `feat/#90`

Docstrings generation was requested by @jaepyo-Lee.

* https://github.com/prography/10th-3team-minari-BE/pull/93#issuecomment-3029518766

The following files were modified:

* `src/main/java/com/prography/minari/answer/service/AnswerService.java`
* `src/main/java/com/prography/minari/aws/impl/FileUploader.java`
* `src/main/java/com/prography/minari/aws/impl/S3Uploader.java`
* `src/main/java/com/prography/minari/common/config/AwsS3Config.java`
* `src/main/java/com/prography/minari/mail/service/MailService.java`

jaepyo-Lee committed:
Merge pull request #109 from prography/coderabbitai/docstrings/c7c5448

📝 Add docstrings to `feat/#90`
